javalib/src/main/scala/java/net/DatagramPacket.scala

```diff
@@ -0,0 +1,89 @@
+package java.net
+
+final class DatagramPacket(
+    var buf: Array[Byte],
+    var offset: Int,
+    var length: Int,
+    // bufLength: Int,
+    var address: InetAddress,
+    var port: Int
+) {
+  if (length < 0 || offset < 0 || (length + offset) < 0 || ((length + offset) > buf.length)) {
+    throw new IllegalArgumentException("illegal length or offset");
```

Comment:
Is this text directly from a JVM?

If yes, then we probably need to live with it.

If no, Conflating three cases into one makes it hard for
users and maintainers to readily know the actual defect.

"illegal" always makes me expect the Sheriff's deputies to
come knock down my door. "Invalid" or such is more descriptive.